confbot_driver/src/nodes/twist_publisher.cpp

```diff
@@ -52,17 +52,34 @@ TwistPublisher::TwistPublisher(rclcpp::NodeOptions options)
       for (auto parameter : parameters) {
         if (parameter.get_name() == "speed") {
           if (rclcpp::ParameterType::PARAMETER_NOT_SET == parameter.get_type()) {
-            RCLCPP_INFO(this->get_logger(),
+            RCLCPP_WARN(this->get_logger(),
               "cannot delete '%s' as it is a required parameter",
               parameter.get_name().c_str()
             );
             result.successful = false;
-          } else {
+          } else if (rclcpp::ParameterType::PARAMETER_DOUBLE == parameter.get_type()) {
             RCLCPP_INFO(this->get_logger(),
               "set parameter '%s' to '%f'",
               parameter.get_name().c_str(),
               parameter.as_double());
             speed_ = parameter.as_double();
+          } else if (rclcpp::ParameterType::PARAMETER_INTEGER == parameter.get_type()) {
+            RCLCPP_INFO(this->get_logger(),
```

Comment:
I'd personally extract the log message here to avoid code duplication. The info message really is the same whether we catch the value as integer or double.

Reply:
that was my initial approach but it adds a nesting level to the conditional making the code harder to read, so I went back to duplicating it. But I don't feel strongly about it